CDCgov/prime-reportstream
# Platform/jessica/12461 json message serialization

This PR changes the messages that we put in the blob store to be JSON.

Leaving old code path in so nothing gets lost that is sitting in the queues, marked the code that needs to be removed with todos, and created this ticket to take care of it: https://app.zenhub.com/workspaces/platform-6182b02547c1130010f459db/issues/gh/cdcgov/prime-reportstream/12614

Test Steps:
1. Make sure the tests pass
2. Test that an HL7 message and a SHIR message are successfully sent. 
3. Switch branch to master, run RS, send a message into the queue, wait for that message to appear in each of the steps that have been updated. Force stop all of the Java processes with Activity Monitor. Switch to this branch. Start up RS. Make sure the messages get sent through successfully. 

## Changes
- Puts JSON in the blob store rather than our own custom, made up format. 

## Checklist

### Testing

### Process

## Linked Issues
#12461 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
